Client/Frontend/BraveVPN/BraveVPNRegionPickerViewController.swift

```diff
@@ -0,0 +1,236 @@
+        }
+        
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    @available(*, unavailable)
+    required init?(coder: NSCoder) { fatalError() }
+    
+    override func viewDidLoad() {
+        title = Strings.VPN.regionPickerTitle
+        
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.register(VPNRegionCell.self)
+        
+        NotificationCenter.default.addObserver(self, selector: #selector(vpnConfigChanged(notification:)),
+                                               name: .NEVPNStatusDidChange, object: nil)
+        
+        view.addSubview(tableView)
+        tableView.snp.makeConstraints {
+            $0.edges.equalToSuperview()
+        }
+    }
+    
+    @objc private func vpnConfigChanged(notification: NSNotification) {
+        guard let connection = notification.object as? NEVPNConnection else { return }
+        
+        if connection.status == .connected {
+            dispatchGroup?.leave()
+        }
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        tableView.reloadData()
+    }
+}
+
+// MARK: - UITableView Data Source & Delegate
+extension BraveVPNRegionPickerViewController: UITableViewDelegate, UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        Section.allCases.count
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        section == Section.automatic.rawValue ? 1 : regionList.count
+    }
+    
+    func tableView(_ tableView: UITableView, titleForFooterInSection section: Int) -> String? {
+        if section == Section.automatic.rawValue {
+            return Strings.VPN.regionPickerAutomaticDescription
+        }
+        
+        return nil
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(for: indexPath) as VPNRegionCell
+        cell.accessoryType = .none
+        
+        switch indexPath.section {
+        case Section.automatic.rawValue:
+            cell.textLabel?.text = Strings.VPN.regionPickerAutomaticModeCellText
+            if Preferences.VPN.vpnRegionOverride.value == nil {
+                cell.accessoryType = .checkmark
+            }
+        case Section.regionList.rawValue:
+            guard let server = regionList[safe: indexPath.row] else { return cell }
+            cell.textLabel?.text = server.namePretty
+            if server.name == Preferences.VPN.vpnRegionOverride.value {
+                cell.accessoryType = .checkmark
+            }
+        default:
+            assertionFailure("Section count out of bounds")
+        }
+        
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
```

Comment:
:o what is the reason we de-select the row? Doesn't tableView automatically de-select it OR are we configuring it somewhere so that the selected row stays selected?

Reply:
It doesn't auto deselect, we do that for almost every tableview we have in the app